example/example_test.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2021 The logr Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
```

Comment:
> As a followup, maybe change the example to use these properly?

Added that as `example/example_test.go`. I'm just not sure whether godoc will pick up the example code there or whether it needs to be one level up in the same package?

Reply:
It turned out that they are not found inside the `example` directory.

It worked for funcr when in the same directory as the code (https://pkg.go.dev/github.com/go-logr/logr@v1.1.0/funcr#pkg-examples) but not for zapr (https://pkg.go.dev/github.com/go-logr/zapr@v1.1.0).

I'll move the file in a new PR.